lib/KmsClient.js

```diff
@@ -703,6 +703,9 @@ export class KmsClient {
       const result = await httpClient.post(url, {
         agent, headers, json: config
       });
+
+      _assert(result.data, 'result.data', 'object');
+      _assert(result.data.id, 'result.data.id', 'string');
```

Comment:
I don't feel like the `_handleClientError` implementation in the catch block is doing to do a good job of surfacing this class of error to the user, I don't thing this is a "client error".  This is a response validation error, so maybe this validation step should happen after we exit this phase in another try/catch block.

Reply:
Also, is it easy to write test vectors for these error condtions?

Reply:
It was too involved for the given fix to implement a bad server for the purpose of testing.